service_config.go

```diff
@@ -229,15 +230,24 @@ type jsonName struct {
 	Method  *string
 }
 
-func (j jsonName) generatePath() (string, bool) {
+var errIgnoreName = errors.New("ignore name")
+
+func (j jsonName) generatePath() (string, error) {
 	if j.Service == nil {
-		return "", false
+		return "", errIgnoreName
```

Comment:
What if j.Service is nil, but j.Method is set to non-empty string?

Reply:
`errIgnoreName` will make it simply ignore the name just like the current implementation does.

https://github.com/grpc/grpc-proto/blob/dd2dca318eb197b96b60f22297871fb1ed862800/grpc/service_config/service_config.proto#L67 states `service` is required, so probably we should return the corresponding error instead of skipping malformed names, but I was afraid to introduce breaking changes.

Reply:
https://github.com/grpc/grpc-go/pull/3684/commits/ef23d5c8d3ff225335aa39165e56ae0f3b2eae00#diff-bd49846390d90e2bd0f528a7a4c1e533R339-R341

I removed this block in the latest patch, let me know if it's not something you wanted me to do.